metrics/firebase/functions/test/main_test.dart

```diff
@@ -450,6 +454,441 @@ void main() {
+
+        await onBuildUpdatedHandler(change, null);
+
+        final projectIdMatcher = predicate<DocumentData>(
+          (data) => data.getString('projectId') == projectId,
+        );
+
+        verify(
+          documentReference.setData(
+            argThat(projectIdMatcher),
+            any,
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "increments a build day document's successful field value if the updated build document snapshot's status is successful",
+      () async {
+        final buildDayStatusFieldName =
+            BuildDayStatusFieldName.successful.value;
+
+        whenAccessChangeData(documentSnapshot: newDocumentSnapshot)
+            .thenReturn(buildDocumentData);
+
+        await onBuildUpdatedHandler(change, null);
+
+        final statusFieldIncrementMatcher =
+            documentFieldIncrementMatcher(buildDayStatusFieldName, 1);
+
+        verify(
+          documentReference.setData(
+            argThat(statusFieldIncrementMatcher),
+            any,
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "increments a build day document's failed field value if the updated build document snapshot's status is failed",
+      () async {
+        final buildDayStatusFieldName = BuildDayStatusFieldName.failed.value;
+        final newBuildJson = testDataGenerator.generateBuildJson(
+          buildStatus: BuildStatus.failed,
+        );
+
+        whenAccessChangeData(documentSnapshot: newDocumentSnapshot)
+            .thenReturn(DocumentData.fromMap(newBuildJson));
+
+        await onBuildUpdatedHandler(change, null);
+
+        final statusFieldIncrementMatcher =
+            documentFieldIncrementMatcher(buildDayStatusFieldName, 1);
+
+        verify(
+          documentReference.setData(
+            argThat(statusFieldIncrementMatcher),
+            any,
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "increments a build day document's unknown field value if the updated build document snapshot's status is unknown",
+      () async {
+        final buildDayStatusFieldName = BuildDayStatusFieldName.unknown.value;
+        final newBuildJson = testDataGenerator.generateBuildJson(
+          buildStatus: BuildStatus.unknown,
+        );
+
+        whenAccessChangeData(documentSnapshot: newDocumentSnapshot)
+            .thenReturn(DocumentData.fromMap(newBuildJson));
+
+        await onBuildUpdatedHandler(change, null);
+
+        final statusFieldIncrementMatcher =
+            documentFieldIncrementMatcher(buildDayStatusFieldName, 1);
+
+        verify(
+          documentReference.setData(
+            argThat(statusFieldIncrementMatcher),
+            any,
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "decrements a build day document's inProgress field value if the old build document snapshot's status is inProgress and there are no tasks with such build id in tasks collection",
```

Comment:
Should we test that this function decrements the corresponding field of the build days document depending on build data before the update?